lua/telescope/entry_manager.lua

```diff
@@ -130,14 +130,16 @@ function EntryManager:add_entry(picker, score, entry)
     return
   end
 
+  -- may need the prompt for tiebreak
+  local prompt = picker:_get_prompt()
   for index, container, node in self.linked_states:ipairs() do
     info.looped = info.looped + 1
 
     if container[2] > score then
       return self:_insert_container_before(picker, index, node, new_container)
     end
 
-    if score < 1 and container[2] == score and #entry.ordinal < #container[1].ordinal then
+    if score < 1 and container[2] == score and picker.tiebreak(entry, container[1], prompt) then
```

Comment:
I would change it to `picker:tiebreak(entry, container[1])`. This preserves maximum flexibility at little to no cost with a transparent signature.

Your `prompt` use case could then with minimal overhead as follows:

```lua
local opts = {}
opts.tiebreak = setmetatable({}, {
  __call = function(self, picker, current_entry, existing_entry)
    if not self.prompt then
      self.prompt = picker:_get_prompt()
    end
    -- .. use prompt in comparison
  end,
})
```

Not that I could come up on the spot why anyone would use more than `prompt` of picker, but I'm sure people will manage :sweat_smile: